xbmc/input/InputCodingTableBaiduPY.cpp

```diff
@@ -121,6 +122,30 @@ std::vector<std::wstring> CInputCodingTableBaiduPY::GetResponse(int response)
   return words;
 }
 
+bool CInputCodingTableBaiduPY::Initialize()
+{
+  CSingleLock lock(m_CS);

```

Comment:
we're protecting Create(), it's not thread safe and if it's called twice during init it calls terminate